### PR TITLE
Move dataset stack deployment code to the analysis runner repository.

### PR DESCRIPTION
**Pull request by lgruen**

### CODE REVIEW

File: stack/Pulumi.fewgenomes.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+encryptionsalt: v1:SPQXlVggjxw=:v1:X1sTpViNuyK+Wcom:3GANI8gZOKtk0gG7BklsyHeNU5uVLw==
+config:
+  datasets:customer_id: C010ys3gt
+  datasets:enable_release: "false"
+  datasets:hail_service_account: fewgenomes-911@hail-295901.iam.gserviceaccount.com
```
Comment (illusional) [inline stack/Pulumi.fewgenomes.yaml:5]: Are these okay to be public facing? I know they're just service accounts.

File: stack/__main__.py [hunk 2]
```diff
@@ -0,0 +1,391 @@
+"""
+Pulumi stack to set up buckets and permission groups
+"""
+
+import pulumi
+import pulumi_gcp as gcp
+
+DOMAIN = 'populationgenomics.org.au'
+CUSTOMER_ID = 'C010ys3gt'
```
Comment (illusional) [inline stack/__main__.py:9]: Another is this okay to be public facing.

File: stack/__main__.py [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+
+
+def group_mail(kind: str) -> str:
+    """Returns the email address of a permissions group."""
+    return f'{dataset}-{kind}@{DOMAIN}'
+
+
+def create_group(mail: str) -> gcp.cloudidentity.Group:
+    """Returns a new Cloud Identity group for the given email address."""
+    name = mail.split('@')[0]
```
Comment (illusional) [inline stack/__main__.py:118]: Should we process the name further, by maybe making it lowercase with `.lower()`?

File: stack/Pulumi.fewgenomes.yaml (hunk 1 above)
Comment (lgruen) [reply stack/Pulumi.fewgenomes.yaml:5]: Thanks for checking! Yes, I don't think somebody else could use this information without having additional permissions.

File: stack/__main__.py (hunk 2 above)
Comment (lgruen) [reply stack/__main__.py:9]: Ditto.

File: stack/__main__.py (hunk 3 above)
Comment (lgruen) [reply stack/__main__.py:118]: The name here is just a Pulumi identifier that's displayed in the plan. So there's no need to do a lot of processing, it just was a bit long when having the domain attached to it.

**Review by illusional — APPROVED** [review]
Comments address my issue, and seems pretty good!

**Pull request merged**